.github/actions/checkout-ref/action.yml

```diff
@@ -0,0 +1,17 @@
+name: Checkout a particular ref
```

Comment:
[actions/checkout@v2](https://github.com/actions/checkout) does appear to provide the option to check out nested repositories. Was that considered as an alternative to this and the `fetch-depth: 0` workaround?

Reply:
It does check them out, but what we want is specify _which ref_ to check out, independent from what the configuration of the main repository says; `actions/checkout@v2` does not provide this functionality...

Reply:
I did not mean the `submodules: true` option -- I meant like this:
```
- name: Checkout
  uses: actions/checkout@v2

- name: Checkout tools repo
  uses: actions/checkout@v2
  with:
    repository: my-org/my-tools
    path: my-tools
```

Reply:
But does it accept a `ref`?

Reply:
I suppose we could _not_ automatically check out submodules but do it manually and provide a ref if one was specified... That might work.

Reply:
I had thought that, but since we already have something that works, I guess we don't need to worry about it.

Reply:
It would amount to a simpler solution though, so I might give that a shot.

Reply:
OK, I found a way to make it work with `actions/checkout@v2`. I'll remove the custom action.